### PR TITLE
Refactor EMPI Delegate and propagate through consumers (V2 pathway)

### PR DESCRIPTION
**Pull request by rayking29**
# Fixes or Implements [AB#15346](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/15346)

## Testing

- [x] Unit Tests Updated
- [ ] Functional Tests Updated
- [ ] Not Required

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

File: Apps/AccountDataAccess/src/Patient/ClientRegistriesDelegate.cs [hunk 1]
```diff
@@ -223,21 +224,15 @@ private static Name ExtractName(PN nameSection)
 
         private void CheckResponseCode(string responseCode)
         {
-            if (responseCode.Contains("BCHCIM.GD.2.0018", StringComparison.InvariantCulture))
-            {
-                // BCHCIM.GD.2.0018 Not found
-                this.logger.LogWarning("Client Registry did not find any records. Returned message code: {ResponseCode}", responseCode);
-                throw new ProblemDetailsException(ExceptionUtility.CreateProblemDetails(ErrorMessages.ClientRegistryRecordsNotFound, HttpStatusCode.NotFound, nameof(ClientRegistriesDelegate)));
-            }
```
Comment (MikeLyttle) [inline Apps/AccountDataAccess/src/Patient/ClientRegistriesDelegate.cs:231]: Encountering this error code is supposed to cause the delegate to return null.

File: Apps/AccountDataAccess/src/Patient/PatientModel.cs [hunk 2]
```diff
@@ -81,5 +72,10 @@ public class PatientModelV2
         /// Gets or sets the response code for the patient.
         /// </summary>
         public string ResponseCode { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets if the patient is deceased.
+        /// </summary>
+        public bool? IsDeceased { get; set; }
```
Comment (MikeLyttle) [inline Apps/AccountDataAccess/src/Patient/PatientModel.cs:79]:
```suggestion
        /// <summary>
        /// Gets or sets a value indicating whether the patient is deceased.
        /// </summary>
        public bool? IsDeceased { get; set; }
```

File: Apps/Patient/src/Controllers/PatientController.cs [hunk 3]
```diff
@@ -102,5 +102,25 @@ public async Task<IActionResult> GetPatientV2(string hdid)
...
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status502BadGateway)]
+        [ApiVersion("3.0")]
+        [Route("{hdid}")]
+        [Authorize(Policy = PatientPolicy.Read)]
+        public async Task<ActionResult<PatientDetails>> GetPatientV3(string hdid)
+        {
+            return this.Ok(await this.serviceV2.GetPatient(hdid).ConfigureAwait(true));
+        }
```
Comment (MikeLyttle) [inline Apps/Patient/src/Controllers/PatientController.cs:124]: I don't think we'll need this, but we can leave it in for now.

File: Apps/Patient/src/Services/IPatientService.cs [hunk 4]
```diff
@@ -31,7 +31,12 @@ public interface IPatientService
...
         /// <param name="disableIdValidation">Disables the validation on HDID/PHN when true.</param>
+        /// <param name="ct">The cancellation token.</param>
         /// <returns>The patient model.</returns>
-        Task<ApiResult<PatientModelV2>> GetPatient(string identifier, PatientIdentifierType identifierType = PatientIdentifierType.Hdid, bool disableIdValidation = false);
+        Task<PatientDetails> GetPatient(
+            string identifier,
+            PatientIdentifierType identifierType = PatientIdentifierType.Hdid,
+            bool disableIdValidation = false,
+            CancellationToken ct = default);
     }
```
Comment (MikeLyttle) [inline Apps/Patient/src/Services/IPatientService.cs:41]: May as well also rename to `GetPatientAsync()`.

**Review by BronzBierd — COMMENTED** [review]
I'm only seeing things that I would clean up when moving such large code, I will leave these for now to allow you to focus on delivering the requirements, I'll add suggestions in the final review. 

Mike has reviewed the error handling so I won't be adding anything at this time.

**Review by MikeLyttle — APPROVED** [review]

**Review by BrianMaki — APPROVED** [review]

**Pull request merged**